examples_tests/01.HelloWorld/main.cpp

```diff
@@ -286,13 +302,236 @@ int main()
+	subpassDescription.preserveAttachmentCount = 0u;
+	subpassDescription.preserveAttachments = nullptr;
+
+	video::IGPURenderpass::SCreationParams renderPassParams;
+	renderPassParams.attachmentCount = 1u;
+	renderPassParams.attachments = &attachmentDescription;
+	renderPassParams.dependencies = nullptr;
+	renderPassParams.dependencyCount = 0u;
+	renderPassParams.subpasses = &subpassDescription;
+	renderPassParams.subpassCount = 1u;
+
+	core::smart_refctd_ptr<video::IGPURenderpass> renderPass = device->createGPURenderpass(renderPassParams);
+
+	std::array<core::smart_refctd_ptr<video::IGPUFramebuffer>, SC_IMG_COUNT> fbos;
+	const auto swapchainImages = swapchain->getImages();
+	const uint32_t swapchainImageCount = swapchain->getImageCount();
+	assert(swapchainImageCount == SC_IMG_COUNT);
+
+	for (uint32_t i = 0u; i < swapchainImageCount; ++i)
+	{
+		auto img = swapchainImages.begin()[i];
+		core::smart_refctd_ptr<video::IGPUImageView> imageView;
+		{
+			video::IGPUImageView::SCreationParams viewParams;
+			viewParams.format = img->getCreationParameters().format;
+			viewParams.viewType = asset::IImageView<video::IGPUImage>::ET_2D;
+			viewParams.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT;
+			viewParams.subresourceRange.baseMipLevel = 0u;
+			viewParams.subresourceRange.levelCount = 1u;
+			viewParams.subresourceRange.baseArrayLayer = 0u;
+			viewParams.subresourceRange.layerCount = 1u;
+			viewParams.image = std::move(img); // this might create problems
+
+			imageView = device->createGPUImageView(std::move(viewParams));
+			assert(imageView);
+		}
+
+		video::IGPUFramebuffer::SCreationParams fbParams;
+		fbParams.width = WIN_W;
+		fbParams.height = WIN_H;
+		fbParams.layers = 1u;
+		fbParams.renderpass = renderPass;
+		fbParams.flags = static_cast<video::IGPUFramebuffer::E_CREATE_FLAGS>(0);
+		fbParams.attachmentCount = 1u; // Todo(achal): must be equal to the corresponding value in render pass
+		fbParams.attachments = &imageView;
+
+		fbos[i] = device->createGPUFramebuffer(std::move(fbParams));
+	}
+
+	const uint32_t FRAMES_IN_FLIGHT = 2u;
+
+	// acquireSemaphore will be signalled once you acquire the image to render
+	// releaseSeamphore will be signalled once you rendered to the image and you're ready to release it to present it
+	core::smart_refctd_ptr<video::IGPUSemaphore> acquireSemaphores[FRAMES_IN_FLIGHT];
+	core::smart_refctd_ptr<video::IGPUSemaphore> releaseSemaphores[FRAMES_IN_FLIGHT];
+	core::smart_refctd_ptr<video::IGPUFence> frameFences[FRAMES_IN_FLIGHT];
+	for (uint32_t i = 0u; i < FRAMES_IN_FLIGHT; ++i)
 	{
+		acquireSemaphores[i] = device->createSemaphore();
+		releaseSemaphores[i] = device->createSemaphore();
+		frameFences[i] = device->createFence(video::IGPUFence::E_CREATE_FLAGS::ECF_SIGNALED_BIT);
 	}
 
+	video::IGPUQueue* graphicsQueue = device->getQueue(graphicsFamilyIndex, 0u);
+	video::IGPUQueue* presentQueue = device->getQueue(presentFamilyIndex, 0u);
+
+	// Todo(achal): Hacky stuff begin, get rid

```

Comment:
because of the circular refs?

Reply:
This example is largely unchanged, please look at example 2 for the most recent implementation